### PR TITLE
Switch Kustomize to use bases and not resources.

### LINKED ISSUE #611: Kustomize should use bases instead of resources
/kind feature

**Describe the solution you'd like**
In an effort to move to support Kustomize2, @figo pointed out that there is a nice way to gain support without breaking changes, while still maintaining backward compatibility. Kustomize supports `bases`, which is a list of folders that can be a relative path (see [cluster-api](https://github.com/kubernetes-sigs/cluster-api/tree/master/config) as reference). 

Each folder will have its own `kustomization.yaml` file referencing the files that should be exported.

/priority important-longterm
/milestone Next

### PR DESCRIPTION
**Pull request by sfzylad**
It will prevent problems while switching to kustomize2.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #611 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: config/BUILD.bazel [hunk 1]
```diff
@@ -34,22 +34,35 @@ controller_gen(
...
     name = "provider-non-stamped-base",
     srcs = [
-        "//config/default:kustomization.yaml",
-        "//config/default:rbac_role_binding_patch.yaml",
-        "//config/manager:manager.yaml",
+        "//config:default/rbac_role_binding_patch.yaml",
+        "//config:manager/manager.yaml",
         "config",
+        "//config:default/kustomization.yaml",
+        #"//config:rbac/kustomization.yaml",
```
Comment (detiber) [inline config/BUILD.bazel]: Is the comment here intentional?

Comment (sfzylad) [reply config/BUILD.bazel]: It suppose to be removed. Fixed in the new commit.

File: config/manager/BUILD.bazel [hunk 2]
```diff
@@ -1,20 +0,0 @@
...
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-exports_files(
-    [
-        "manager.yaml",
-    ],
-    visibility = ["//visibility:public"],
-)
```
Comment (vincepri) [inline config/manager/BUILD.bazel]: Any reason why this bazel file got removed?

File: config/default/BUILD.bazel [hunk 3]
```diff
@@ -1,21 +0,0 @@
...
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-exports_files(
-    [
-        "kustomization.yaml",
-        "rbac_role_binding_patch.yaml",
-    ],
-    visibility = ["//visibility:public"],
-)
```
Comment (vincepri) [inline config/default/BUILD.bazel]: Same here

File: config/manager/BUILD.bazel (hunk 2 above)
Comment (sfzylad) [reply config/manager/BUILD.bazel]: As far as I remember it clashed with the entries I added to `config/BUILD.bazel` and removing it fixed it.

File: config/default/BUILD.bazel (hunk 3 above)
Comment (sfzylad) [reply config/default/BUILD.bazel]: As far as I remember it clashed with the entries I added to `config/BUILD.bazel` and removing it fixed it.

**Comment by sfzylad** [comment]
There's one more small change on the way. Please don't approve it yet.

**Comment by sfzylad** [comment]
It seems ready for review.

**Comment by randomvariable** [comment]
Will do the review for tomorrow. Sorry about the delay.

**Pull request merged**